### PR TITLE
Add git resolver

### PR DESCRIPTION
**Pull request by magik6k**
Depends on https://github.com/ipld/js-ipld-git/pull/1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "ipld-dag-cbor": "~0.11.1",
     "ipld-dag-pb": "~0.11.0",
     "ipld-eth-star": "^1.4.1",
+    "ipld-git": "file:../js-ipld-git",
```
Comment (diasdavid) [inline package.json]: woa, this one is new to me. I never saw a `file:` in a package.json. This is almost like gx-rw

Comment (diasdavid) [reply package.json]: Once the changes requested in ipld-git are done, let's update this line

**Comment by diasdavid** [comment]
@magik6k tests are not happy, wanna check?

**Comment by magik6k** [comment]
Travis fails on node 4 as it doesn't support spread operator (https://travis-ci.org/ipld/js-ipld-resolver/jobs/271251389#L3638)

Circle is.. interesting, looks like `ipldGit.util.cid` returns invalid CID or that error assert is ignored (which it probably isn't). It passes for me locally. My only theory is that old multicodec package is cached here. @diasdavid would that make sense?

**Comment by diasdavid** [comment]
> Travis fails on node 4 as it doesn't support spread operator (https://travis-ci.org/ipld/js-ipld-resolver/jobs/271251389#L3638)

Going to remove the support for it in Master and then you can rebase.

> My only theory is that old multicodec package is cached here. @diasdavid would that make sense?

When that happens, just give it a kick :) 

image

**Comment by diasdavid** [comment]
@magik6k ok, you can rebase now https://github.com/ipld/js-ipld-resolver/commit/d9069c88f9c0a65d119192a77fd95219977dd9d0

**Comment by magik6k** [comment]
@diasdavid CI looks happier

File: src/index.js [hunk 2]
```diff
@@ -64,6 +65,10 @@ class IPLDResolver {
                      dagCBOR.resolver,
                      dagCBOR.util)
 
+    this.support.add(ipldGit.resolver.multicodec,
+                     ipldGit.resolver,
+                     ipldGit.util)
```
Comment (diasdavid) [inline src/index.js]: 🌟🌟🌟🌟🌟🌟

**Pull request merged**